src/app/appReducers.js

```diff
@@ -38,12 +39,20 @@ export default (state: AppState = initialState, action: Action) => {
         ...state,
         needsInitialFetch: !!action.apiKey,
       };
-    case REHYDRATE:
+    case REHYDRATE: {
+      const needsInitialFetch = !!getAuth(action.payload).apiKey;
+      if (action.payload.chat && needsInitialFetch) {
+        // note: Reassigning a variable reference is not mutation
```

Comment:
Thanks for the link, so I can see you are wrong 😄 
Reassigning a variable like
```js
state = undefined;
```
does not mutate anything, that is true.

Doing this though:
```js
state.something = undefined;
```
You are mutating `state`, and you are not mutating `something`.